src/core/ticker/Ticker.js

```diff
@@ -217,18 +221,86 @@ export default class Ticker
      *
      * @param {Function} fn - The listener function to be added for one update
      * @param {Function} [context] - The listener context
+     * @param {number} [priority=10] - The priority for emitting
+     * @returns {PIXI.ticker.Ticker} This instance of a ticker
+     */
+    addOnce(fn, context, priority = Ticker.PRIORITY_DEFAULT)
+    {
+        return this._add(fn, context, priority, true);
+    }
+
+    /**
+     * Internally adds the event handler so that it can be sorted by priority.
+     * Priority allows certain handler (user, AnimatedSprite, Interaction) to be run
+     * before the rendering.
+     *
+     * @private
+     * @param {Function} fn - The listener function to be added for one update
+     * @param {Function} context - The listener context
+     * @param {number} priority - The priority for emitting
+     * @param {boolean} once - If it should only be added once
      * @returns {PIXI.ticker.Ticker} This instance of a ticker
      */
-    addOnce(fn, context)
+    _add(fn, context, priority, once)
     {
-        this._emitter.once(TICK, fn, context);
+        const listeners = this._listeners;
+        const emitter = this._emitter;
+
+        listeners.push(new TickerListener(fn, context, priority, once));
+
+        // Sort by prority from highest to lowest
+        listeners.sort((listener1, listener2) => listener2.priority - listener1.priority);
```

Comment:
No the point was performance, but...
I never actually tested performance of this with the splice, it actually negates the perf as seen here:
https://jsperf.com/insertion-sort-with-copy/1

I'm fine with your implementation as you have it since it already makes the array copy. If you can find any way to use my method without the copy up front that's great too ;) You can see that without the copy, the splice method blows the native sort away. Too bad.